File: win/directx/sdk-9.0/Bin/DXUtils/Visual Studio 7.0 Wizards/C#/VC#Wizards/Mdx9AppWizard/Templates/1033/ddraw.cs
using System;
using System.Windows.Forms;
using System.Drawing;
using Microsoft.DirectX;
using Microsoft.DirectX.DirectDraw;

/// <summary>
/// This class is where the Drawing routines
/// for non-Direct3D and non-DirectDraw 
/// applications reside.
/// </summary>
public class GraphicsClass
{
    private Control owner = null;

[!if !VISUALTYPE_BLANK]
    private const int spriteSize = 50;
[!endif]
[!if VISUALTYPE_BLANK]
    private Surface surfaceText = null;
[!endif]
    private Device localDevice = null;
    private Clipper localClipper = null;
    private Surface surfacePrimary = null;
    private Surface surfaceSecondary = null;

    public GraphicsClass(Control owner)
    {
        this.owner = owner;

        localDevice = new Device();
        localDevice.SetCooperativeLevel(owner, CooperativeLevelFlags.Normal);
        
        CreateSurfaces();
    }

    private void CreateSurfaces()
    {
        SurfaceDescription desc = new SurfaceDescription();
        SurfaceCaps caps = new SurfaceCaps(); 

        localClipper = new Clipper(localDevice);
        localClipper.Window = owner;

        desc.SurfaceCaps.PrimarySurface = true;
        surfacePrimary = new Surface(desc, localDevice);
        surfacePrimary.Clipper = localClipper;
        
        desc.Clear();
        desc.SurfaceCaps.OffScreenPlain = true;
        desc.Width = surfacePrimary.SurfaceDescription.Width;
        desc.Height = surfacePrimary.SurfaceDescription.Height;
        
        surfaceSecondary = new Surface(desc, localDevice);            
[!if !VISUALTYPE_BLANK]
        surfaceSecondary.FillStyle = 0;
[!endif]

[!if VISUALTYPE_BLANK]
        desc.Clear();
        desc.SurfaceCaps.OffScreenPlain = true;
        desc.Width = 100;
        desc.Height = 50;
        surfaceText = new Surface(desc, localDevice);
        surfaceText.ForeColor = Color.Black;
[!endif]
    }

[!if !VISUALTYPE_BLANK]
    public void RenderGraphics(Point destination)
    {
        if (!owner.Created)
            return;

        Rectangle dest = new Rectangle(owner.PointToScreen(new Point(destination.X, destination.Y)),new Size(spriteSize, spriteSize));

        if (null == surfacePrimary || null == surfaceSecondary)
            return;

        try
        {
            surfaceSecondary.ColorFill(Color.Blue);
            surfaceSecondary.DrawEllipse(dest.Left, dest.Y, dest.Right, dest.Bottom);
            surfacePrimary.Draw(surfaceSecondary, DrawFlags.DoNotWait);
        }
        catch(SurfaceLostException)
        {
            // The surface can be lost if power saving
            // mode kicks in, or any other number of
            // reasons.
            CreateSurfaces();
        }
    }
[!endif]
[!if VISUALTYPE_BLANK]
    public void RenderGraphics(string text)
    {
        Rectangle r = new Rectangle(new Point(owner.Left, owner.Top), new Size(surfacePrimary.SurfaceDescription.Width, surfacePrimary.SurfaceDescription.Height));

        if (!owner.Created) 
            return;

        if (null == surfacePrimary || null == surfaceSecondary)
            return;

        try
        {
            surfaceSecondary.ColorFill(Color.Blue);
            surfaceText.ColorFill(Color.Blue);
            surfaceText.DrawText(0, 0, text, true);
            surfaceSecondary.Draw(new Rectangle(10, 40, 100, 50), surfaceText, DrawFlags.DoNotWait);
            surfacePrimary.Draw(r, surfaceSecondary, DrawFlags.DoNotWait);
        }
        catch(SurfaceLostException)
        {
            // The surface can be lost if power saving
            //mode kicks in, or any other number of
            //reasons.
            CreateSurfaces();
        }
    }
[!endif]
}

File: win/directx/sdk-9.0/Bin/DXUtils/Visual Studio 7.0 Wizards/VB .NET/VBWizards/Mdx9AppWizard/Templates/1033/ddraw.vb
Imports System
Imports System.Windows.Forms
Imports System.Drawing
Imports Microsoft.DirectX
Imports Microsoft.DirectX.DirectDraw

'/ <summary>
'/ This class is where the Drawing routines
'/ for non-Direct3D and non-DirectDraw 
'/ applications reside.
'/ </summary>
Public Class GraphicsClass
    Private owner As Control = Nothing

[!if !VISUALTYPE_BLANK]
    Private Const spriteSize As Integer= 50
[!endif]
[!if VISUALTYPE_BLANK]
    Private surfaceText As Surface = Nothing
[!endif]
    Private localDevice As Device = Nothing
    Private localClipper As Clipper = Nothing
    Private surfacePrimary As Surface = Nothing
    Private surfaceSecondary As Surface = Nothing

    Public Sub New(ByRef owner As Control)
        Me.owner = owner

        localDevice = New Device()
        localDevice.SetCooperativeLevel(owner, CooperativeLevelFlags.Normal)
        
        CreateSurfaces()
    End Sub

    Private Sub CreateSurfaces()
        Dim desc As SurfaceDescription = New SurfaceDescription()
        Dim caps As SurfaceCaps = New SurfaceCaps() 

        localClipper = New Clipper(localDevice)
        localClipper.Window = owner

        desc.SurfaceCaps.PrimarySurface = true
        surfacePrimary = New Surface(desc, localDevice)
        surfacePrimary.Clipper = localClipper
        
        desc.Clear()
        desc.SurfaceCaps.OffScreenPlain = true
        desc.Width = surfacePrimary.SurfaceDescription.Width
        desc.Height = surfacePrimary.SurfaceDescription.Height
        
        surfaceSecondary = New Surface(desc, localDevice)            
[!if !VISUALTYPE_BLANK]
        surfaceSecondary.FillStyle = 0
[!endif]

[!if VISUALTYPE_BLANK]
        desc.Clear()
        desc.SurfaceCaps.OffScreenPlain = True
        desc.Width = 100
        desc.Height = 50
        surfaceText = New Surface(desc, localDevice)
        surfaceText.ForeColor = Color.Black
[!endif]
    End Sub

[!if !VISUALTYPE_BLANK]
    Public Sub RenderGraphics(ByVal destination As Point)
        If (Not owner.Created) Then
            Return
        End If

        Dim dest As Rectangle = New Rectangle(owner.PointToScreen(New Point(destination.X, destination.Y)), New Size(spriteSize, spriteSize))

        If (Nothing Is surfacePrimary Or Nothing Is surfaceSecondary) Then
            Return
        End If

        Try
            surfaceSecondary.ColorFill(Color.Blue)
            surfaceSecondary.DrawEllipse(dest.Left, dest.Top, dest.Right, dest.Bottom)
            surfacePrimary.Draw(surfaceSecondary, DrawFlags.DoNotWait)
        Catch e As SurfaceLostException
            ' The surface can be lost if power saving
            ' mode kicks in, or any other number of
            ' reasons.
            CreateSurfaces()
        End Try
    End Sub
[!else]
    Public Sub RenderGraphics(ByVal text As String)
        Dim r As New Rectangle(New Point(owner.Left, owner.Top), New Size(surfacePrimary.SurfaceDescription.Width, surfacePrimary.SurfaceDescription.Height))

        If (Not owner.Created) Then
            Return
        End If

        If (Nothing Is surfacePrimary Or Nothing Is surfaceSecondary) Then
            Return
        End If

        Try
            surfaceSecondary.ColorFill(Color.Blue)
            surfaceText.ColorFill(Color.Blue)
            surfaceText.DrawText(0, 0, text, True)
            surfaceSecondary.Draw(New Rectangle(10, 40, 100, 50), surfaceText, DrawFlags.DoNotWait)
            surfacePrimary.Draw(r, surfaceSecondary, DrawFlags.DoNotWait)
        Catch e As SurfaceLostException
            ' The surface can be lost if power saving
            ' mode kicks in, or any other number of
            ' reasons.
            CreateSurfaces()
        End Try
    End Sub
[!endif]
End Class